diff --git a/python3/httplib2/certs.py b/python3/httplib2/certs.py
index 59d1ffc..31b8dfe 100644
--- a/python3/httplib2/certs.py
+++ b/python3/httplib2/certs.py
@@ -13,7 +13,7 @@ except ImportError:
 custom_ca_locater_available = False
 custom_ca_locater_where = None
 try:
-    from ca_certs_locater import get as custom_ca_locater_where
+    from .ca_certs_locater import get as custom_ca_locater_where
     custom_ca_locater_available = True
 except ImportError:
     pass
